kroxylicious-kms-providers/kroxylicious-kms-provider-azure-key-vault-kms/src/test/java/io/kroxylicious/kms/provider/azure/keyvault/KeyVaultClientIT.java

```diff
@@ -0,0 +1,142 @@
+import org.testcontainers.images.PullPolicy;
+import org.testcontainers.utility.DockerImageName;
+
+import com.azure.security.keyvault.keys.KeyClient;
+import com.azure.security.keyvault.keys.cryptography.CryptographyClient;
+import com.azure.security.keyvault.keys.cryptography.models.KeyWrapAlgorithm;
+import com.azure.security.keyvault.keys.cryptography.models.WrapResult;
+import com.azure.security.keyvault.keys.models.CreateKeyOptions;
+import com.azure.security.keyvault.keys.models.KeyOperation;
+import com.azure.security.keyvault.keys.models.KeyType;
+import com.azure.security.keyvault.keys.models.KeyVaultKey;
+import com.github.nagyesta.lowkeyvault.testcontainers.LowkeyVaultClientFactory;
+import com.github.nagyesta.lowkeyvault.testcontainers.LowkeyVaultContainer;
+
+import io.kroxylicious.kms.provider.azure.WrappingKey;
+import io.kroxylicious.kms.provider.azure.auth.BearerToken;
+import io.kroxylicious.kms.provider.azure.auth.BearerTokenService;
+import io.kroxylicious.kms.provider.azure.config.AzureKeyVaultConfig;
+import io.kroxylicious.kms.provider.azure.config.auth.EntraIdentityConfig;
+import io.kroxylicious.proxy.config.secret.InlinePassword;
+import io.kroxylicious.proxy.config.tls.InsecureTls;
+import io.kroxylicious.proxy.config.tls.Tls;
+
+import static com.github.nagyesta.lowkeyvault.testcontainers.LowkeyVaultContainerBuilder.lowkeyVault;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.when;
+
+@ExtendWith(MockitoExtension.class)
+class KeyVaultClientIT {
+
+    public static final String VAULT_NAME = "default";
+    public static final Tls INSECURE_TLS = new Tls(null, new InsecureTls(true), null, null);
+    public static final String KEY_NAME = "my-key";
+
+    @Mock
+    BearerTokenService bearerTokenService;
+
+    LowkeyVaultContainer startVault() {
+        // bit annoying to have to sync the image name
+        final DockerImageName imageName = DockerImageName.parse("nagyesta/lowkey-vault:4.0.0");
+        final LowkeyVaultContainer lowkeyVaultContainer = lowkeyVault(imageName)
+                .vaultNames(Set.of(VAULT_NAME))
+                .build()
+                .withImagePullPolicy(PullPolicy.defaultPolicy());
+        lowkeyVaultContainer.start();
+        return lowkeyVaultContainer;
+    }
+
+    @Test
+    void getKey() {
+        try (LowkeyVaultContainer lowkeyVaultContainer = startVault()) {
+            BearerToken token = new BearerToken("mytoken", Instant.MIN, Instant.MAX);
+            when(bearerTokenService.getBearerToken()).thenReturn(CompletableFuture.completedFuture(token));
+            EntraIdentityConfig unused = new EntraIdentityConfig(null, "tenant", new InlinePassword("abc"), new InlinePassword("def"), null, null);
+            KeyVaultClient keyVaultClient = new KeyVaultClient(bearerTokenService,
+                    new AzureKeyVaultConfig(unused, lowkeyVaultContainer.getVaultBaseUrl(VAULT_NAME), INSECURE_TLS));
```

Comment:
`lowkeyVaultContainer.getVaultBaseUrl(VAULT_NAME)` fails for me because it generates a url with a hostname that doesn't resolve.

<img width="2454" height="1381" alt="image" src="https://github.com/user-attachments/assets/56b19273-b1a0-4da5-abac-acc453862ce8" />